src/RA_AchievementSet.cpp

```diff
@@ -420,148 +423,142 @@ BOOL AchievementSet::LoadFromFile(GameID nGameID)
     const std::string sFilename = GetAchievementSetFilename(nGameID);
 
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pFile = nullptr;
-    errno_t nErr = fopen_s(&pFile, sFilename.c_str(), "r");
-    if (pFile != nullptr)
-    {
-        //	Store this: we are now assuming this is the correct checksum if we have a file for it
-        g_pCurrentGameData->SetGameID(nGameID);
+    std::ifstream ifile{ sFilename };
 
-        if (m_nSetType == Local)
-        {
-            const char EndLine = '\n';
-            char buffer[4096];
-            unsigned long nCharsRead = 0;
+    // in this case we do have to check
+    if (!ifile.is_open())
+        return FALSE;
 
-            //	Get min ver:
-            _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-            //	UNUSED at this point? TBD
 
-            //	Get game title:
-            _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-            if (nCharsRead > 0)
-            {
-                buffer[nCharsRead - 1] = '\0';	//	Turn that endline into an end-string
+    //	Store this: we are now assuming this is the correct checksum if we have a file for it
+    g_pCurrentGameData->SetGameID(nGameID);
 
-                //	Loading Local from file...?
-                g_pCurrentGameData->SetGameTitle(buffer);
-            }
-
-            while (!feof(pFile))
+    if (m_nSetType == Local)
+    {
+        std::string buffer;
+        buffer.reserve(4096); 
```

Comment:
This is required. If you don't think it is try it for yourself. Since we're doing a binary operation on the file we need a pointer, the "capacity" is the same thing as memory in bytes except we don't have to worry about deallocating since `std::string` is managing the pointer for us.